system/lib/gl.c

```diff
@@ -1559,174 +1597,279 @@ void* emscripten_GetProcAddress(const char *name_) {
+  else if (!strcmp(name, "eglGetProcAddress")) return emscripten_GetProcAddress;
+  else if (!strcmp(name, "glActiveTexture")) return emscripten_glActiveTexture;
+  else if (!strcmp(name, "glAttachShader")) return emscripten_glAttachShader;
+  else if (!strcmp(name, "glBeginQuery")) return emscripten_glBeginQuery;
+  else if (!strcmp(name, "glBeginTransformFeedback")) return emscripten_glBeginTransformFeedback;
+  else if (!strcmp(name, "glBindAttribLocation")) return emscripten_glBindAttribLocation;
+  else if (!strcmp(name, "glBindBuffer")) return emscripten_glBindBuffer;
+  else if (!strcmp(name, "glBindBufferBase")) return emscripten_glBindBufferBase;
+  else if (!strcmp(name, "glBindBufferRange")) return emscripten_glBindBufferRange;
+  else if (!strcmp(name, "glBindFramebuffer")) return emscripten_glBindFramebuffer;
+  else if (!strcmp(name, "glBindRenderbuffer")) return emscripten_glBindRenderbuffer;
+  else if (!strcmp(name, "glBindSampler")) return emscripten_glBindSampler;
   else if (!strcmp(name, "glBindTexture")) return emscripten_glBindTexture;
-  else if (!strcmp(name, "glGetTexParameterfv")) return emscripten_glGetTexParameterfv;
-  else if (!strcmp(name, "glGetTexParameteriv")) return emscripten_glGetTexParameteriv;
-  else if (!strcmp(name, "glTexParameterfv")) return emscripten_glTexParameterfv;
-  else if (!strcmp(name, "glTexParameteriv")) return emscripten_glTexParameteriv;
-  else if (!strcmp(name, "glIsTexture")) return emscripten_glIsTexture;
-  else if (!strcmp(name, "glGenBuffers")) return emscripten_glGenBuffers;
-  else if (!strcmp(name, "glDeleteBuffers")) return emscripten_glDeleteBuffers;
-  else if (!strcmp(name, "glGetBufferParameteriv")) return emscripten_glGetBufferParameteriv;
+  else if (!strcmp(name, "glBindTransformFeedback")) return emscripten_glBindTransformFeedback;
+  else if (!strcmp(name, "glBindVertexArray")) return emscripten_glBindVertexArray;
+  else if (!strcmp(name, "glBlendColor")) return emscripten_glBlendColor;
+  else if (!strcmp(name, "glBlendEquation")) return emscripten_glBlendEquation;
+  else if (!strcmp(name, "glBlendEquationSeparate")) return emscripten_glBlendEquationSeparate;
+  else if (!strcmp(name, "glBlendFunc")) return emscripten_glBlendFunc;
+  else if (!strcmp(name, "glBlendFuncSeparate")) return emscripten_glBlendFuncSeparate;
+  else if (!strcmp(name, "glBlitFramebuffer")) return emscripten_glBlitFramebuffer;
   else if (!strcmp(name, "glBufferData")) return emscripten_glBufferData;
   else if (!strcmp(name, "glBufferSubData")) return emscripten_glBufferSubData;
-  else if (!strcmp(name, "glIsBuffer")) return emscripten_glIsBuffer;
-  else if (!strcmp(name, "glGenRenderbuffers")) return emscripten_glGenRenderbuffers;
+  else if (!strcmp(name, "glCheckFramebufferStatus")) return emscripten_glCheckFramebufferStatus;
+  else if (!strcmp(name, "glClear")) return emscripten_glClear;
+  else if (!strcmp(name, "glClearBufferfi")) return emscripten_glClearBufferfi;
```

Comment:
The function `glClearBufferfi` (`emscripten_glClearBufferfi`) is a WebGL 2 function. If user is targeting WebGL 1, this function would be undefined. Also if co-targeting both WebGL 1 and WebGL 2 dynamically (e.g. page chooses which one to initialize based on browser availability at page startup), this should return null function pointer if the context version was just GLES2/WebGL 1.